docs/css-style-guide.md

```diff
@@ -0,0 +1,92 @@
+Hypothesis CSS Style Guide
+==========================
+
+Hypothesis' CSS generally follows the principles of [Trello's CSS Style Guide](https://github.com/trello/trellisheets)
+for writing maintainable styling. In particular:
+
+ * Use simple, non-nested class selectors where possible
+ * Organize CSS into components. Unlike Trello, we use BEM-style naming (see below) and do not indent selectors
```

Comment:
BEM-styling naming has the slight advantage that you can parse a class name into independently re-usable block/component and elements which only make sense within their containing block by looking just at the name.

eg. Looking at `.tab-bar__tab--dark` you can tell that "tab-bar" is a re-usable thing, 'tab' is a sub-element of it that you wouldn't use outside of a tab bar.

I've moved the Trello reference just to the Further Reading section now so it is clear that it provides guidance rather than a template for this guide.